### PR TITLE
Vocpupuli pull withcommits

### PR DESCRIPTION
**Pull request by b4ldr**
@bastelfreak i think most of your comments where to do with me using retrospec puppet to create a bunch of boilerplate stuff.  I have no removed that and just added the changes i have made to the types, spec, readme and fixtures files.

Currently this is not building do to 

```
ruby_dep-1.5.0 requires ruby version >= 2.2.5, which is incompatible with the
```

i fixed this in my Gemfile by adding

``` ruby
gem "listen", "<=3.0.6"
```

however after looking at your example the following may be better

``` ruby
gem "listen", "<=3.0.6" if :require => false RUBY_VERSION < '2.2.5'
```

however im not an expert in ruby and im not familiar with plugin sync so i thought i would leave this for now.  let me know if there is anything else you want to see.  i have closed the other pull request however i have left the branch on my repo incase something was missed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#### Table of Contents
```
Comment (domcleal) [inline README.md:1]: This file's already here as README.markdown, can you remove the old one? (Git/GitHub should automatically see the rename.)

File: .fixtures.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+fixtures:
+    symlinks:
+      tea:  "#{source_dir}"
+      test: "#{source_dir}/spec/fixtures/test"
+    repositories:
+      stdlib:
```
Comment (domcleal) [inline .fixtures.yml:6]: Is this required? I can't see where.

File: spec/aliases/puppetcontent_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          [ nil ],
+          [ nil, nil ],
+          { 'foo' => 'bar' },
+          { },
+          '',
+          "C:/whatever",
+          "\\var\\tmp",
+          "\\Users/hc/wksp/stdlib",
+          "*/Users//nope"
+          "/Users//nope"
```
Comment (domcleal) [inline spec/aliases/puppetcontent_spec.rb]: missing commas here, causing a syntax error in tests

**Review by domcleal — CHANGES_REQUESTED** [review]

**Comment by domcleal** [comment]
Some of the new types are now in stdlib's master branch, are you sure they should be here too? (https://github.com/puppetlabs/puppetlabs-stdlib/commit/83539371e5b1e6ff412eda48c3a725653c506ece)

File: .fixtures.yml (hunk 2 above)
Comment (b4ldr) [reply .fixtures.yml:6]: to you mean stdlib? its possible it is not needed i think i just added without thinking

**Comment by b4ldr** [comment]
Yes i noticed some of them in stdlib.  The reason i have added them here is 
  1) there has not been a release of stdlib which includes those types so i can not use them in production environments
  2)  The old validate functions have been deprecated so even when stdlib releases a new version it will be far to noisey to use in my current environment as there are many many modules which will start throwing warnings.  

I will maintain my own branch regardless so happy to remove the duplicates if thats the prfrerence

**Comment by roidelapluie** [comment]
@domcleal @b4ldr if stdlib gets types now, then this module is useless. We should just merge our types to stdlib

**Comment by bastelfreak** [comment]
@roidelapluie I am not sure about that approach. stdlib always needs ages for new releases. I would be happy if we could keep the types here that aren't in stdlib.

**Comment by roidelapluie** [comment]
@bastelfreak «stdlib always needs ages for new releases» that is NOT a good point.

In this regard, I think keeping puppet-tea is BAD. Instead of VoxPupuli putting everything in its namespace, VP members should learn to contribute to existing repositories instead of creating their own.

File: .fixtures.yml (hunk 2 above)
Comment (domcleal) [reply .fixtures.yml:6]: Yes, stdlib - it can probably be removed, there's no dependency on it that I can see.

**Comment by juniorsysadmin** [comment]
@roidelapluie @b4ldr @bastelfreak @domcleal Given a number of types from puppet-tea are now in stdlib 4.13.1 is this Pull Request still needed?

**Comment by b4ldr** [comment]
Im happy for this to get closed.  I havn't had a chance to move to the latest stdlib myself but once i do ill create a pull request there

**Pull request closed without merging**